clients/js/algorand.ts

```diff
@@ -0,0 +1,164 @@
+import {
+  CONTRACTS,
+  getIsTransferCompletedAlgorand,
+  redeemOnAlgorand,
+} from "@certusone/wormhole-sdk";
+import { NETWORKS } from "./networks";
+import { impossible, Payload } from "./vaa";
+import {
+  Account,
+  Algodv2,
+  assignGroupID,
+  mnemonicToSecretKey,
+  waitForConfirmation,
+} from "algosdk";
+import { TransactionSignerPair } from "@certusone/wormhole-sdk/lib/cjs/algorand";
+
+async function signSendAndConfirmAlgorand(
+  algodClient: Algodv2,
+  txs: TransactionSignerPair[],
+  wallet: Account
+) {
+  assignGroupID(txs.map((tx) => tx.tx));
+  const signedTxns: Uint8Array[] = [];
+  for (const tx of txs) {
+    if (tx.signer) {
+      signedTxns.push(await tx.signer.signTxn(tx.tx));
+    } else {
+      signedTxns.push(tx.tx.signTxn(wallet.sk));
+    }
+  }
+  await algodClient.sendRawTransaction(signedTxns).do();
+  const result = await waitForConfirmation(
+    algodClient,
+    txs[txs.length - 1].tx.txID(),
+    1
+  );
+  return result;
+}
+
+export async function execute_algorand(
+  payload: Payload,
+  vaa: Uint8Array,
+  environment: "MAINNET" | "TESTNET" | "DEVNET"
+) {
+  const ALGORAND_HOST =
+    environment === "MAINNET"
+      ? {
+          algodToken: "",
+          algodServer: "https://mainnet-api.algonode.cloud",
```

Comment:
Should this be in alogrand.rpc in networks.ts?

Reply:
Partially